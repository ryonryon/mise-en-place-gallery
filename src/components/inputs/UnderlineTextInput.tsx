import styled from "@emotion/styled";
import * as React from "react";

import InputType from "./input-type";
import { secondary } from "../../constants/colors";

const StyledInput = styled.input`
  height: 24px;
  width: 70%;
  margin: 16px;
  padding: 4px;
  border: none;
  border-bottom: 1px solid ${secondary};
  color: ${secondary};

  &::placeholder {
    color: ${secondary};
  }
`;

interface Props extends React.InputHTMLAttributes<HTMLInputElement> {
  type: InputType;
}

const UnderlineTextInput = ({ type, ...rest }: Props) => <StyledInput type="text" {...rest} />;

UnderlineTextInput.defaultProps = {
  type: InputType.TEXT
};

export default UnderlineTextInput;
