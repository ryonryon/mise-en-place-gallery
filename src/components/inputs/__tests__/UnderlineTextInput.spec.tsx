import * as React from "react";
import * as renderer from "react-test-renderer";

import UnderlineTextInput from "../UnderlineTextInput";

describe("UnderlineTextInput test", () => {
  it("tests UnderlineTextInput renders correctly ", () => {
    const underlineTextInput = renderer.create(<UnderlineTextInput />).toJSON();
    expect(underlineTextInput).toMatchSnapshot();
  });
});
