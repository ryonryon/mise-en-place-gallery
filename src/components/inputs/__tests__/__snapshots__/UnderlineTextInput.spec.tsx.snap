// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnderlineTextInput test tests UnderlineTextInput renders correctly  1`] = `
<input
  className="css-1ym2sj7"
  type="text"
/>
`;
